geopandas/kvec.h

```diff
@@ -0,0 +1,90 @@
+/* The MIT License
```

Comment:
Can you put at the top of this file a short note about why this file is includes (what it provides (implementation of resizeable vector) and why we need it (used to store matching indices in the spatial join implementation in algos.c)